packages/@aws-cdk/aws-codestarnotifications/lib/notification-rule.ts

```diff
@@ -0,0 +1,237 @@
+import { IResource, Resource, Names } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnNotificationRule } from './codestarnotifications.generated';
+import { NotificationRuleSourceConfig, INotificationRuleSource } from './source';
+import { INotificationRuleTarget, NotificationRuleTargetConfig } from './target';
+
+/**
+ * The level of detail to include in the notifications for this resource.
+ */
+export enum DetailType {
+  /**
+   * BASIC will include only the contents of the event as it would appear in AWS CloudWatch
+   */
+  BASIC = 'BASIC',
+
+  /**
+   * FULL will include any supplemental information provided by AWS CodeStar Notifications and/or the service for the resource for which the notification is created.
+   */
+  FULL = 'FULL',
+}
+
+/**
+ * The status of the notification rule.
+ */
+enum Status {
+
+  /**
+   * If the status is set to DISABLED, notifications aren't sent.
+   */
+  DISABLED = 'DISABLED',
+
+  /**
+   * If the status is set to ENABLED, notifications are sent.
+   */
+  ENABLED = 'ENABLED',
+}
+
+/**
+ * Standard set of options for `notifyOnXxx` codestar notification handler on construct
+ */
+export interface NotifyOptions {
```

Comment:
This name is a little confusing to me. Let's rename this to `NotificationRuleOptions`.

Reply:
Changed.